### PR TITLE
fix(typeahead): fix typeahead's breaking if typeahead property receives NULL

### PR DESCRIPTION
**Pull request by ghost**
Closes to #4417

### CODE REVIEW

File: src/typeahead/typeahead.directive.ts [hunk 1]
```diff
@@ -448,6 +448,9 @@ export class TypeaheadDirective implements OnInit, OnDestroy {
   }
 
   protected finalizeAsyncCall(matches: TypeaheadMatch[]): void {
+    if (!matches) {
+      return;
```
Comment (valorkin) [inline src/typeahead/typeahead.directive.ts]:
```suggestion
       matches = [];
```

File: demo/src/app/components/+typeahead/demos/async/async.ts [hunk 2]
```diff
@@ -78,6 +78,9 @@ export class DemoTypeaheadAsyncComponent {
 
   getStatesAsObservable(token: string): Observable<any> {
     const query = new RegExp(token, 'i');
+    if (!this.statesComplex) {
```
Comment (valorkin) [inline demo/src/app/components/+typeahead/demos/async/async.ts]: remove after testing

**Review by valorkin — CHANGES_REQUESTED** [review]

File: demo/src/app/components/+typeahead/demos/async/async.ts (hunk 2 above)
Comment (ghost) [reply demo/src/app/components/+typeahead/demos/async/async.ts]: removed

File: src/typeahead/typeahead.directive.ts (hunk 1 above)
Comment (ghost) [reply src/typeahead/typeahead.directive.ts]: fixed in other way

**Review by valorkin — APPROVED** [review]

**Pull request merged**